python/src/spark_rapids_ml/core.py

```diff
@@ -1505,6 +1507,13 @@ def predict_udf(iterator: Iterator[pd.DataFrame]) -> Iterator[pd.Series]:
         pred_col = predict_udf(struct(*select_cols))
 
         if self._is_single_pred(dataset.schema):
+            if "array" in output_schema:
```

Comment:
Seems better to check the type of pred_col specifically since that is what will be transformed potentially.

Reply:
Yeah, the type of pred_col is obtained through output_schema. Is there a way to get type from the pred_col Column variable?
 
Revised the code to handle the case when output_schema is StructType.

Reply:
Can you look up the type of pred_col specifically in the output_schema vs just checking if there is an array type in output_schema?

Reply:
The output_schema of PCA is "array<float>" without the column name. Seems there is no way to lookup through pred column name. Is there any other way?

Reply:
Note `self._is_single_pred(dataset.schema)` is True so there is only one column in the output_schema. Seems ok to check the "array<float>" and "array<double>" keyword.